src/fps/proposal/MultisigProposal.sol

```diff
@@ -0,0 +1,854 @@
+    struct Call3Value {
+        address target;
+        bool allowFailure;
+        uint256 value;
+        bytes callData;
+    }
+
+    /// @notice Task TOML config file values
+    struct TaskConfig {
+        string[] allowedStorageWriteAccesses;
+        string description;
+        string name;
+        string safeAddressString;
+    }
+
+    /// @notice configuration set at initialization
+    TaskConfig public config;
+
+    /// @notice flag to determine if the proposal is being simulated
+    bool private _buildStarted;
+
+    /// @notice buildModifier to be used by the build function to populate the
+    /// actions array
+    modifier buildModifier() {
+        require(multisig != address(0), "Must set addresses object for multisig address to be set");
+
+        require(!_buildStarted, "Build already started");
+        _buildStarted = true;
+
+        _startBuild();
+        _;
+        _endBuild();
+
+        _buildStarted = false;
+    }
+
+    /// @notice Initialize the proposal with task and network configuration
+    /// @param taskConfigFilePath Path to the task configuration file
+    /// @param networkConfigFilePath Path to the network configuration file
+    /// @param _addresses Address registry contract
+    function _init(string memory taskConfigFilePath, string memory networkConfigFilePath, Addresses _addresses)
+        internal
+    {
+        require(
+            !initialized && bytes(config.safeAddressString).length == 0 && address(addresses) == address(0x0),
+            "MultisigProposal: already initialized"
+        );
+        setTaskConfig(taskConfigFilePath);
+        setL2NetworksConfig(networkConfigFilePath, _addresses);
+        initialized = true;
+    }
+
+    /// @notice Set the task configuration
+    /// @param taskConfigFilePath Path to the task configuration file
+    function setTaskConfig(string memory taskConfigFilePath) public override {
+        require(
+            block.chainid == getChain("mainnet").chainId || block.chainid == getChain("sepolia").chainId,
+            string.concat("Unsupported network: ", vm.toString(block.chainid))
+        );
+
+        string memory taskConfigFileContents = vm.readFile(taskConfigFilePath);
+
+        bytes memory fileContents = vm.parseToml(taskConfigFileContents, ".task");
+        config = abi.decode(fileContents, (TaskConfig));
+
+        name = config.name;
+        description = config.description;
+    }
+
+    /// @notice Sets the L2 networks configuration
+    /// @param networkConfigFilePath Path to the network configuration file
+    /// @param _addresses Address registry contract
+    function setL2NetworksConfig(string memory networkConfigFilePath, Addresses _addresses) public override {
+        addresses = _addresses;
+        string memory networkConfigFileContents = vm.readFile(networkConfigFilePath);
+
+        nonce = abi.decode(vm.parseToml(networkConfigFileContents, ".safeNonce"), (uint256));
```

Comment:
IIRC `parseToml` will fail if the nonce is not defined right? Just curious what happens in this case.

Reply:
Sounds like a good test case :)

Reply:
Awesome - I'll resolve this thread and look for more testing in a later PR.